ios/Classes/BetterPlayer.m

```diff
@@ -238,6 +238,12 @@ - (void)setDataSourcePlayerItem:(AVPlayerItem*)item withKey:(NSString*)key{
     _playerRate = 1;
     [_player replaceCurrentItemWithPlayerItem:item];
 
+    if (!@available(iOS 16.0, *)) {
+         // set buffer time into 5 seconds
+         // https://dw-ml-nfc.atlassian.net/browse/DAF-3642
+         item.preferredForwardBufferDuration = 5;
```

Comment:
I think we might need to adjust this value for preferredForwardBufferDuration.
Is it possible do test again?
(Please consider there is 10 second seek forward by tapping right side of player. )

It looks the `5` means minimum buffering duration. It can still buffer more than that.
I check the buffered duration by this logging
```
    _betterPlayerController.addEventsListener((event) {
      if (event.betterPlayerEventType == BetterPlayerEventType.progress) {
        debugPrint(
            'value.buffered: ${_betterPlayerController.videoPlayerController?.value.buffered}, position: ${_betterPlayerController.videoPlayerController?.value.position}, duration: ${_betterPlayerController.videoPlayerController?.value.duration}, ');
      }
    });
```


When I tested with `5`, some time buffered period becomes longer 10second (max: 15sec) and when do seek to 10 second forward (by tapping right side of screen), there was issue.
```
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:00.418000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:01.019000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:01.618000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:02.218000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:02.818000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:03.418000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:04.018000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:04.617000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:05.218000, duration: 0:03:51.000000, // Here buffer 10 sec when buffered period becoming less than 5 second.
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:05.818000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:06.418000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:07.018000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:07.618000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:08.219000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:08.818000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:09.418000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:10.018000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:10.618000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:11.218000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:11.817000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:12.418000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:13.019000, duration: 0:03:51.000000,
```

When I tested `1`, it keep buffer max 10sec (safer than `5`)
```
lutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:00.000000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:00.732000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:01.332000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:01.932000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:02.531000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:03.132000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:03.732000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:04.332000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:04.933000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:05.532000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:06.133000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:06.732000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:07.331000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:07.932000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:09.933000)], position: 0:00:08.533000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:08.832000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:09.733000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:10.332000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:10.932000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:11.533000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:12.132000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:12.733000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:13.333000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:13.931000, duration: 0:03:51.000000,
flutter: value.buffered: [DurationRange(start: 0:00:00.000000, end: 0:00:19.933000)], position: 0:00:14.234000, duration: 0:03:51.000000,
````

Reply:
I tested with 10 seconds buffer time and it works well, thank you for your suggestion @HiroshiOshiro 🙇